pkg/kf/commands/quotas/integration_test.go

```diff
@@ -0,0 +1,184 @@
+package quotas
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	. "github.com/GoogleCloudPlatform/kf/pkg/kf/testutil"
+)
+
+// TestIntegration_Create creates a resourcequota, then tries to retrieve the quota info.
+// It finally deletes the quota.
+func TestIntegration_Create(t *testing.T) {
+	checkClusterStatus(t)
+	RunKfTest(t, func(ctx context.Context, t *testing.T, kf *Kf) {
+		quotaName := fmt.Sprintf("integration-quota-%d", time.Now().UnixNano())
+		defer kf.DeleteQuota(ctx, quotaName)
+
+		memQuantity := "500Gi"
+		cpuQuantity := "10"
+		routesQuantity := "100"
+
+		createQuotaOutput := kf.CreateQuota(ctx, quotaName,
+			"-m", memQuantity,
+			"-c", cpuQuantity,
+			"-r", routesQuantity,
+		)
+
+		AssertContainsAll(t, strings.Join(createQuotaOutput, " "), []string{quotaName, "successfully created"})
+		Logf(t, "done ensuring quota was created.")
+
+		// Get the quota
+		getQuotaOutput := kf.GetQuota(ctx, quotaName)
+		AssertContainsAll(t, strings.Join(getQuotaOutput, " "), []string{quotaName, memQuantity, cpuQuantity, routesQuantity})
+		Logf(t, "done ensuring correct quota info returned.")
+	})
+}
+
+// TestIntegration_Delete creates a quota and then deletes it. It then makes
+// sure the quota no longer exists.
+func TestIntegration_Delete(t *testing.T) {
+	checkClusterStatus(t)
+	RunKfTest(t, func(ctx context.Context, t *testing.T, kf *Kf) {
+		quotaName := fmt.Sprintf("integration-quota-%d", time.Now().UnixNano())
+
+		memQuantity := "500Gi"
+		cpuQuantity := "10"
+		routesQuantity := "100"
+
+		kf.CreateQuota(ctx, quotaName,
+			"-m", memQuantity,
+			"-c", cpuQuantity,
+			"-r", routesQuantity,
+		)
+
+		getQuotaOutput := kf.GetQuota(ctx, quotaName)
+		AssertContainsAll(t, strings.Join(getQuotaOutput, " "), []string{quotaName, memQuantity, cpuQuantity, routesQuantity})
+
+		deleteQuotaOutput := kf.DeleteQuota(ctx, quotaName)
+		AssertContainsAll(t, strings.Join(deleteQuotaOutput, " "), []string{quotaName, "successfully deleted"})
+
+		// Getting the quota should output an error saying the quota is not found
+		getQuotaOutput = kf.GetQuota(ctx, quotaName)
+		AssertContainsAll(t, strings.Join(getQuotaOutput, " "), []string{quotaName, "Error", "not found"})
+	})
+}
+
+// TestIntegration_Update creates a quota then updates it.
+// It checks that the quota is saved with the new values.
+// Afterwards, the quota is deleted.
+func TestIntegration_Update(t *testing.T) {
+	checkClusterStatus(t)
+	RunKfTest(t, func(ctx context.Context, t *testing.T, kf *Kf) {
+		quotaName := fmt.Sprintf("integration-quota-%d", time.Now().UnixNano())
+		defer kf.DeleteQuota(ctx, quotaName)
+
+		memQuantity := "500Gi"
+		cpuQuantity := "10"
+
+		kf.CreateQuota(ctx, quotaName,
+			"-m", memQuantity,
+			"-c", cpuQuantity,
+		)
+		// Check that routes quota is defaulted to "0"
+		getQuotaOutput := kf.GetQuota(ctx, quotaName)
+		AssertContainsAll(t, strings.Join(getQuotaOutput, "\n"), []string{quotaName, memQuantity, cpuQuantity, "0"})
```

Comment:
could be worth noting what this extra "0" you're searching for is, and I think it'll always be true anyway since memQuantity and cpuQuantity both contain it

Reply:
ah good point, I updated the quantities to all be distinct from each other